lib/ukcpio/cpio.c

```diff
@@ -58,6 +58,44 @@ int uk_syscall_r_utime(const char *, const struct utimbuf *);
 int uk_syscall_r_mkdir(const char *, mode_t);
 int uk_syscall_r_symlink(const char *, const char *);
 int uk_syscall_r_stat(const char *, struct stat *);
+int uk_syscall_r_unlinkat(int, const char *, int);
+int uk_syscall_r_rename(const char *, const char *);
+
+static int
+try_rm_nonempty_dir(const char *path)
+{
+	/* rm -rf is nontrivial and costly, so we rename for speed */
+	/* TODO: revisit if efficient recursive dir removal is available */
+	int r;
+	char newpath[PATH_MAX];
+	char *newend = newpath + strlcpy(newpath, path, PATH_MAX);
+
+	if (unlikely(newend - newpath + 2 > PATH_MAX)) {
+		uk_pr_err("Cannot rename %s, path too long\n", path);
+		return -ENAMETOOLONG;
+	}
+	strcpy(newend, ".0");
+	r = uk_syscall_r_rename(path, newpath);
+	uk_pr_info("rename '%s' to '%s': %d\n", path, newpath, r);
+	return r;
+}
+
+static int
+try_remove(const char *path)
+{
+	int r;
+
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, 0);
+	uk_pr_info("unlink %s: %d\n", path, r);
```

Comment:
Ditto. Let's keep all debug messages aligned with other existing ones (starting with capital).

There are others below as well (like `rmdir`), won't spam `ditto` for each one though :grin: